coretex/entities/tag.py

```diff
@@ -0,0 +1,144 @@
+#     Copyright (C) 2023  Coretex LLC
+
+#     This file is part of Coretex.ai
+
+#     This program is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU Affero General Public License as
+#     published by the Free Software Foundation, either version 3 of the
+#     License, or (at your option) any later version.
+
+#     This program is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU Affero General Public License for more details.
+
+#     You should have received a copy of the GNU Affero General Public License
+#     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from enum import IntEnum
+from typing import Optional, Dict, Any
+from abc import abstractmethod, ABC
+
+import re
+import random
+import logging
+
+from ..networking import networkManager, NetworkRequestError
+
+
+class EntityTagType(IntEnum):
+
+    model   = 1
+    dataset = 2
+
+
+class Taggable(ABC):
+
+    id: int
+    projectId: int
+
+    @property
+    @abstractmethod
+    def entityTagType(self) -> int:
+        pass
+
+    def _getTagId(self, tagName: str) -> Optional[int]:
+        parameters = {
+            "name": tagName,
+            "type": int(self.entityTagType),
+            "project_id": self.projectId,
```

Comment:
Why is `project_id` needed to fetch tags?

Reply:
This way only tags of the relevant project are returned. We then check if the inputted tag already exists for that project.